File: src/main/java/com/github/zametki/component/form/PasswordField.java
package com.github.zametki.component.form;

import org.apache.wicket.markup.html.form.PasswordTextField;
import org.apache.wicket.model.IModel;

/**
 * Password field without default wicket validation.
 */
public class PasswordField extends PasswordTextField {

    public PasswordField(String id) {
        super(id);
        setRequired(false);
    }

    public PasswordField(String id, IModel<String> model) {
        super(id, model);
        setRequired(false);
    }
}

File: src/main/java/com/github/zametki/component/form/PasswordField.kt
package com.github.zametki.component.form

import org.apache.wicket.markup.html.form.PasswordTextField
import org.apache.wicket.model.IModel

/**
 * Password field without default wicket validation.
 */
class PasswordField(id: String, model: IModel<String>) : PasswordTextField(id, model) {

    init {
        isRequired = false
    }
}
